encodings/fastlanes/src/bitpacking/mod.rs

```diff
@@ -122,6 +123,16 @@ impl BitPackedArray {
             .vortex_expect("BitpackedArray is missing packed child bytes array")
     }
 
+    #[inline]
+    pub fn packed_slice<T: NativePType + BitPacking>(&self) -> &[T] {
+        let packed_primitive = self.packed().as_primitive();
+        let maybe_null_slice = packed_primitive.maybe_null_slice::<T>();
+        // SAFETY: maybe_null_slice points to buffer memory that outlives the lifetime of `self`.
+        //  Unfortunately Rust cannot understand this, so we reconstruct the slice from raw parts
+        //  to get it to reinterpret the lifetime.
+        unsafe { std::slice::from_raw_parts(maybe_null_slice.as_ptr(), maybe_null_slice.len()) }
```

Comment:
i think i'm right about this but would appreciate another pair of eyes.

If I don't do this reinterpret, and instead just do

```
return self.packed().as_primitive().maybe_null_slice()
```

it says that you cannot return reference to an intermediate value. the slice is obviously a ref though to the underlying data buffer.